examples/remote-sensing/plot_clustering.py

```diff
@@ -0,0 +1,373 @@
+    """
+
+    def __init__(self, window_size: int, overlap: int = 0):
+        assert window_size % 2 == 1, 'Window size must be odd.'
+        assert overlap >= 0, 'Overlap must be positive.'
+        assert overlap <= window_size//2,\
+                'Overlap must be smaller or equal than int(window_size/2).'
+        self.window_size = window_size
+        self.overlap = overlap
+
+    def fit(self, X: ArrayLike, y = None):
+        return self
+
+    def transform(self, X: ArrayLike):
+        X = sliding_window_view(
+                X,
+                window_shape=(self.window_size, self.window_size),
+                axis=(0, 1))
+        if self.overlap is not None:
+            if self.overlap > 0:
+                X = X[::self.overlap, ::self.overlap]
+        else:
+            X = X[::self.window_size//2, ::self.window_size//2]
+            self.overlap = self.window_size//2
+
+        # Reshape to (n_pixels, n_samples, n_features) with n_pixels=axis0*axis1
+        # n_samples=axis3*axis_4 and n_features=axis2
+        X = X.reshape((-1, X.shape[2], X.shape[3]*X.shape[4]))
+        return X
+        
+    def fit_transform(self, X: ArrayLike, y=None):
+        return self.fit(X).transform(X)
+
+
+class LabelsToImage(BaseEstimator, TransformerMixin):
+    """Predicted labels to image taking into account sliding windows.
+
+    Parameters
+    ----------
+    height : int
+        Height of the original image.
+
+    width : int
+        Width of the original image.
+
+    window_size : int
+        Size of the window.
+    """
+
+    def __init__(self, height: int, width: int,
+                 window_size: int, overlap: int = 0):
+        assert window_size % 2 == 1, 'Window size must be odd.'
+        assert overlap >= 0, 'Overlap must be positive.'
+        assert overlap <= window_size//2, \
+                'Overlap must be smaller or equal than int(window_size/2).'
+        self.height = height
+        self.width = width
+        self.overlap = overlap
+        self.window_size = window_size
+
+    def fit(self, X: ArrayLike, y=None):
+        return self
+
+    def transform(self, X: ArrayLike):
+        # Compute reshape size thanks ot window-size before overlap
+        height = self.height - self.window_size + 1
+        width = self.width - self.window_size + 1
+        # Taking into account overlap
+        if self.overlap > 0:
+            height = ceil(height/self.overlap) 
+            width = ceil(width/self.overlap)
+
+        # Reshape to (height, weight)
+        return X.reshape((height, width))
+
+    def fit_transform(self, X: ArrayLike, y=None):
+        return self.fit(X).transform(X)
+
+
+def pca_image(image: ArrayLike, nb_components: int):
+    """ A function that centers data and applies PCA on an image.
+
+    Parameters
+    ----------
+        image : ArrayLike
+            An image of shape (n_rows, n_columns, n_features).
+
+        nb_components : int
+            Number of components to keep.
```

Comment:
Shape of array is indicated in the first line.
```suggestion
    image : ndarray, shape (n_rows, n_columns, n_features) 
        An image.
    n_components : int
        Number of principal components to keep.
```

Reply:
features is quite broad depending on the application. It can be:
1.  Spectral dimension if several radar were used (very few datasets have this in radar because it require different sensors)
2. Polarimetric information (several modes of polarisation of emitted waves exist and often are used)
3. Interferometric information (sometimes there are two or more acquisition at the same time but with a slightly different looking angle)
4. Time dimension when several acquisitions are done at different times
5. Also depending on the direction the satellite saw the scene there are usually two modes (ascending: satellite going to north direction, descending the reverse). The same satellite can usually do both at slightly different times due to how the orbits work 

In all these cases, the covariance descriptor is useful for classification and the framework isn't so different for the basic processings: computing covariance over spatial coordinates on a sliding window approach (the goal is to classify pixels over a feature dimension. Depending on the used data, the goals and result of classification may differ).

Fo a basics on SAR, see for example: https://elib.dlr.de/82313/1/SAR-Tutorial-March-2013.pdf

Also besides SAR, there are also non-radar optical images (such as the hyper spectral case in the example), where `features`  refers to often spectral dimensions but there can also be temporal.

To conclude, the applications in remote sensing are so large that the data dimensionalities differ vastly: it can go from a simple (n_rows, n_columns) to (n_rows, n_columns, n_looking_angles, n_polarisations, n_spectral_bands, 2 (asc and desc), n_times) to the most extreme case.
Depending on the application, one can consider what they want as feature and the rest can be seen as batch dimensions.